p2p/conn/secret_connection.go

```diff
@@ -118,19 +111,18 @@ func MakeSecretConnection(conn io.ReadWriteCloser, locPrivKey crypto.PrivKey) (*
 	if err != nil {
 		return nil, err
 	}
-	if _, err := io.ReadFull(hkdfHiEphPub, hdkfState[:]); err != nil {
-		return nil, err
-	}
 
-	hkdfSecret := hkdf.New(hash, dhSecret[:], hdkfState[:], []byte("TENDERMINT_SECRET_CONNECTION_TRANSCRIPT_HASH"))
-	if _, err := io.ReadFull(hkdfSecret, hdkfState[:]); err != nil {
-		return nil, err
-	}
+	transcript.AppendMessage([]byte("DH_SECRET"), dhSecret[:])
 
 	// Generate the secret used for receiving, sending, challenge via HKDF-SHA2
 	// on the transcript state (which itself also uses HKDF-SHA2 to derive a key
 	// from the dhSecret).
-	recvSecret, sendSecret, challenge := deriveSecretAndChallenge(hdkfState, locIsLeast)
+	recvSecret, sendSecret := deriveSecrets(dhSecret, locIsLeast)
+
+	var challenge [32]byte
```

Comment:
please avoid using magic numbers.

Is there a constant for this? `ChallengeSize` or something

if not, could you add one

```suggestion
	const challengeSize = 32
	var challenge [challengeSize]byte
```